### PR TITLE
Obey GP's Diagnosis Room Choice

### PR DESCRIPTION
**Pull request by lewri**
Just a starting note -- I am not good with the action queue, so likely this change is a little cautious or not optimal?

**Describe what the proposed change does**
- Fixes the issue where the GP chose a specific diagnosis room for the patient, then the action queue threw that out the window.
- Seek room action now obeys the GP's requested diagnosis room for the patient, if that room is available
- Some additional comments added

### CODE REVIEW

File: CorsixTH/Lua/humanoid_actions/seek_room.lua [hunk 1]
```diff
@@ -188,7 +198,7 @@ local function action_seek_room_start(action, humanoid)
     humanoid:finishAction()
     return
   end
-  -- Tries to find the room, if it is a diagnosis room, try to find any room in the diagnosis list.
+  -- Try to find the appropriate room for the patient
   if not humanoid.diagnosed or action.room_type == "research" then
     local room = action_seek_room_find_room(action, humanoid)
```
Comment (lewri) [inline CorsixTH/Lua/humanoid_actions/seek_room.lua:203]: Think the `local` can be dropped from this line now? Just want someone to confirm.

Comment (lewri) [reply CorsixTH/Lua/humanoid_actions/seek_room.lua:203]: Thanks github actions,for confirming

Comment (lewri) [reply CorsixTH/Lua/humanoid_actions/seek_room.lua:203]: (I actually meant line 68)

File: CorsixTH/Lua/humanoid_actions/seek_room.lua [hunk 2]
```diff
@@ -48,12 +48,22 @@ function SeekRoomAction:setDiagnosisRoom(room)
...
 end
 
+--! Finds a relevant diagnosis/treatment room(s) for a patient
+--! If diagnosis room, attempt to use GP's choice first, else select any other available
+--! room at random.
 local action_seek_room_find_room = permanent"action_seek_room_find_room"( function(action, humanoid)
   local room_type = action.room_type
   if action.diagnosis_room then
+    -- Attempt to use the GP's chosen diagnosis room first
+    local room
```
Comment (TheCycoONE) [inline CorsixTH/Lua/humanoid_actions/seek_room.lua:58]: No need to expand room's scope like this, you can have two `local room` in the same function when the scope doesn't overlap.

Comment (lewri) [reply CorsixTH/Lua/humanoid_actions/seek_room.lua:58]: You could try moving it to line 60 to avoid the warning perhaps?

Comment (Alberth289346) [reply CorsixTH/Lua/humanoid_actions/seek_room.lua:58]: Correct.

`room_at_index` declaration could also be moved more inward.

**Comment by lewri** [comment]
Scope issues fixed

**Comment by Alberth289346** [comment]
Out of scope, but I wonder if the "available diagnosis rooms" of the patient gets extended when I build a new diagnosis room.

**Pull request merged**